apisix/admin/services.lua

```diff
@@ -47,15 +43,15 @@ local function check_conf(id, conf, need_id)
         return nil, {error_msg = "wrong service id"}
     end
 
-
     core.log.info("schema: ", core.json.delay_encode(core.schema.service))
     core.log.info("conf  : ", core.json.delay_encode(conf))
     local ok, err = core.schema.check(core.schema.service, conf)
     if not ok then
         return nil, {error_msg = "invalid configuration: " .. err}
     end
 
-    if need_id and not tonumber(id) then
+    local id_match, _ = ngx_re.match(id, "^[0-9a-zA-Z]{1,}$", "jo")
```

Comment:
I found that this is not a general solution to this problem.

Because not only `service` id has this problem.

So I create another PR: https://github.com/apache/incubator-apisix/pull/1739

You can submit your test cases after #1739 merged.